RELEASE.md

```diff
@@ -0,0 +1,20 @@
+# Plugin broker release process
+
+Che plugin broker releases are built automatically from webhooks on the `release` branch and uses the version specified in `VERSION` to tag the built images. To create a new release:
```

Comment:
Should we link to the CI job, like in other projects?

Reply:
:+1: Good idea